command/write.go

```diff
@@ -147,10 +148,83 @@ func (c *WriteCommand) Run(args []string) int {
 	}
 
 	if secret != nil && secret.Auth != nil && secret.Auth.MFARequirement != nil {
+		if len(secret.Auth.MFARequirement.MFAConstraints) == 1 && !c.flagNonInteractive {
+			// Currently, if there is only one MFA method configured, the login
+			// request is validated interactively
+			methodID, usePasscode := c.getMethodIDUsePasscode(secret.Auth.MFARequirement.MFAConstraints)
+			if methodID != "" {
+				return c.validateMFA(secret.Auth.MFARequirement.MFARequestID, methodID, usePasscode)
+			}
+		}
 		c.UI.Warn(wrapAtLength("A login request was issued that is subject to "+
 			"MFA validation. Please make sure to validate the login by sending another "+
 			"request to mfa/validate endpoint.") + "\n")
 	}
+
+	// Handle single field output
+	if c.flagField != "" {
+		return PrintRawField(c.UI, secret, c.flagField)
+	}
+
+	return OutputSecret(c.UI, secret)
+}
+
+func (c *WriteCommand) getMethodIDUsePasscode(mfaConstraintAny map[string]*logical.MFAConstraintAny) (string, bool) {
+	for _, mfaConstraint := range mfaConstraintAny {
+		if len(mfaConstraint.Any) != 1 {
+			return "", false
+		}
+		return mfaConstraint.Any[0].ID, mfaConstraint.Any[0].UsesPasscode
+	}
+	return "", false
+}
+
+func (c *WriteCommand) validateMFA(reqID, mfaMethodID string, usePasscode bool) int {
+	var passcode string
+	var err error
+	if usePasscode {
+		passcode, err = c.UI.AskSecret(fmt.Sprintf("Enter the passphrase for methodID %q:", mfaMethodID))
```

Comment:
It would be nice if we could tell them the method type, not sure if we have that info.